src/helpers.tsx

```diff
@@ -2,28 +2,47 @@
 import { WrapperSettings } from '@storybook/addons';
 import { DEFAULT_PADDING } from './constants';
 
-type Option = {
-  name: string;
-  value: string;
-  default?: boolean;
+type Size = { name: string; value: string };
+
+type Option = Size & {default?: boolean}
+
+type Sizes = { [key: string]: Size };
+
+type PaddingConfig = {
+  sizes: Sizes;
+  defaultPadding: string;
 }
 
-type Options = Option[] | WrapperSettings['parameters'];
+type Options = Option[] | PaddingConfig | WrapperSettings['parameters'];
+
+export const normalizeEntries = (options: Options) => (Array.isArray(options)
+  ? options
```

Comment:
We should display a deprecation warning in this case, to help users migrate to the new API. We can use Storybook's logger for this, see [example.](https://github.com/storybookjs/storybook/blob/badece61154f4cd372404786186125a7369e362f/addons/backgrounds/src/containers/BackgroundSelector.tsx#L122-L126)